prow/gangway/gangway.proto

```diff
@@ -0,0 +1,129 @@
+  rpc CreateJobExecution(CreateJobExecutionRequest) returns (JobExecution) {
+    option (google.api.http) = {
+      post: "/v1/executions/{job_name}"
+      body: "*"  // See
+                 // https://cloud.google.com/endpoints/docs/grpc/transcoding#use_wildcard_in_body
+    };
+  }
+  rpc GetJobExecution(GetJobExecutionRequest) returns (JobExecution) {
+    // Client example:
+    //   curl http://DOMAIN_NAME/v1/executions/1
+    option (google.api.http) = {
+      get: "/v1/executions/{id}"
+    };
+  }
+  rpc ListJobExecutions(ListJobExecutionsRequest) returns (JobExecutions) {
+    // Client example:
+    //   curl
+    //   http://DOMAIN_NAME/v1/executions?job_name=my-prow-job&status=SUCCESS
+    option (google.api.http) = {
+      get: "/v1/executions"
+    };
+  }
+}
+
+message CreateJobExecutionRequest {
+  string job_name = 1;
+  JobExecutionType job_execution_type = 2;
+  Refs refs = 3;
+  PodSpecOptions pod_spec_options = 4;
+}
+
+message PodSpecOptions {
+  map<string, string> envs = 1;
+  map<string, string> labels = 2;
+  map<string, string> annotations = 3;
+}
+
+/* Look up a single Prow Job execution. */
+message GetJobExecutionRequest {
+  string id = 1;
+}
+
+/* Look up all Prow Job executions that match all fields given here. */
+message ListJobExecutionsRequest {
+  string job_name = 1;            // Mapped to URL query parameter `job_name`.
+  JobExecutionStatus status = 2;  // Mapped to URL query parameter `status`.
+}
+
+message JobExecutions {
+  repeated JobExecution job_execution = 1;
+}
+
+message JobExecution {
+  string id = 1;
+  CreateJobExecutionRequest create_job_execution_request = 2;
+  JobExecutionStatus status = 3;
+  JobExecutionType type = 4;
+  string gcs_path = 5;
+  google.protobuf.Timestamp create_time = 6;
+  google.protobuf.Timestamp completion_time = 7;
+}
+
+// JobExecutionStatus is a 1:1 translation of the existing "ProwJobState" type
+// in prow/apis/prowjobs/v1/types.go.
+enum JobExecutionStatus {
+  JOB_EXECUTION_STATUS_UNSPECIFIED = 0;
+  TRIGGERED = 1;
+  PENDING = 2;
+  SUCCESS = 3;
+  FAILURE = 4;
+  ABORTED = 5;
+  ERROR = 6;
+}
+
+// JobExecutionType is a 1:1 translation of the existing "ProwJobType" type
+// in prow/apis/prowjobs/v1/types.go.
+enum JobExecutionType {
+  JOB_EXECUTION_TYPE_UNSPECIFIED = 0;
+  PERIODIC = 1;
+  POSTSUBMIT = 2;
+  PRESUBMIT = 3;
+  BATCH = 4;
+}
+
+/* Refs is a direct, 1:1 translation of the existing "Refs" struct defined in
+ * prow/apis/prowjobs/v1/types.go.
+ */
+message Refs {
+  string org = 1;
+  string repo = 2;
+  string repo_link = 3;
+  string base_ref = 4;
+  string base_sha = 5;
+  string base_link = 6;
+  repeated Pull pulls = 7;
+  string path_alias = 8;
+  bool work_dir = 9;
+  string clone_uri = 10;
+  bool skip_submodules = 11;
+  int32 clone_depth = 12;
+  bool skip_fetch_head = 13;
+}
+
+/* Pull is a direct, 1:1 translation of the existing "Pull" struct defined in
+ * prow/apis/prowjobs/v1/types.go.
+ */
+message Pull {
```

Comment:
1:1 mapping is great, but I think you had previously mentioned a desire to rename "pull" to something a little more canonical like "pullrequest" or "pr". Is that still something you'd like to do to make the API fields a little more human readable?

Reply:
Well, given how it's 1:1, I would prefer that everything is also named the same way to avoid any confusion.

Reply:
I should say, confusion for Prow developers. For Prow API consumers, I expect the Gangway client lib to hide away some of these naming details so it should be fine.